control/controlclient/direct.go

```diff
@@ -1204,3 +1208,22 @@ func ipForwardingBroken(routes []netaddr.IPPrefix) bool {
 
 	return false
 }
+
+func answerPing(c *http.Client, targetURL string) {
+	ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+	defer cancel()
+
+	req, err := http.NewRequestWithContext(ctx, "HEAD", targetURL, nil)
+	if err != nil {
+		log.Printf("answerPing: %v", err)
+		return
+	}
+	t0 := time.Now()
+	_, err = c.Do(req)
+	d := time.Since(t0).Round(time.Millisecond)
+	if err != nil {
+		log.Printf("answerPing error: %v to %v (after %v)", err, targetURL, d)
+	} else {
+		log.Printf("answerPing complete to %v (after %v)", targetURL, d)
```

Comment:
If we want to use this as a regular part of the protocol we'll want to not log in typical happy case. Maybe the ping request should signal whether or not to log? Maybe KeepAlive==true => no log on success?

Reply:
I can make a new PingRequest struct. A bit overkill considering how much we currently log on success, though. :)

But yeah, we may want control over it to intentionally put stuff in the logs during debugging.